clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -106,16 +118,27 @@ const char *SYCL::Linker::constructLLVMLinkCommand(
   // an actual object/archive.  Take that list and pass those to the linker
   // instead of the original object.
   if (JA.isDeviceOffloading(Action::OFK_SYCL)) {
-    auto SYCLDeviceLibIter =
-        std::find_if(InputFiles.begin(), InputFiles.end(), [](const auto &II) {
-          StringRef InputFilename =
-              llvm::sys::path::filename(StringRef(II.getFilename()));
-          if (InputFilename.startswith("libsycl-") &&
-              InputFilename.endswith(".o"))
-            return true;
-          return false;
-        });
-    bool LinkSYCLDeviceLibs = (SYCLDeviceLibIter != InputFiles.end());
+    auto isSYCLDeviceLib = [](const InputInfo &II) {
+      StringRef InputFilename =
+          llvm::sys::path::filename(StringRef(II.getFilename()));
+      if (!InputFilename.startswith("libsycl-") ||
+          !InputFilename.endswith(".o") || (InputFilename.count('-') < 2))
```

Comment:
For Windows, the objects are represented with `.obj`, right?  We should be checking for that here as well.

Reply:
Hi, @mdtoguchi 
In llvm-link phase, all the input device library input modules are from the output  of clang-offload-bundler, the input of clang-offload-bundler are .o on Linux and .obj on Windows but the extracted llvm IR modules both have the ".o" postfix.
So we only need to check ".o" postfix here.
Thanks very much.

Reply:
Hi, @mdtoguchi 
I realized what you meant, thanks for reminder.

Reply:
Hi, @mdtoguchi 
I have updated the code, if compiler is running on Windows with CL mode, the postfix ".obj" is used.
If compiler is running on Windows with clang mode(CXX, CC mode), the postfix ".o" is used.
Thanks very much.